### PR TITLE
feat: show file sizes

### LINKED ISSUE #253: Show file sizes as part of the download section
Some files are huge (e.g., 5.8 GB for the full CSV file for Colombia) the users should be aware of before downloading the files they want.

### PR DESCRIPTION
**Pull request by yolile**
closes #253

### CODE REVIEW

File: data_registry/templates/includes/files.html [hunk 1]
```diff
@@ -6,14 +6,14 @@
     {% if files.full %}
     <li>
         <a href="{% url 'download_export' %}?spider={{ collection.source_id }}&job_id={{ job.id }}&full=true&suffix={{ suffix }}" rel="nofollow" download>
-            {% translate "All time" %}
+            {% translate "All time" %} ({{ files.full|filesizeformat }})
```
Comment (jpmckinney) [inline data_registry/templates/includes/files.html:9]: Put the file size outside the link. Maybe use `<span class="text-muted">`.

**Comment by yolile** [comment]
image
This is how it looks with the span outside the link.

File: exporter/util.py [hunk 2]
```diff
@@ -195,8 +195,8 @@ def files(self) -> Dict:
                 continue
             prefix = path.name[:4]  # year or "full"
             if prefix.isdigit() and "_" not in path.name:  # don't return month files
-                files[suffix]["years"].add(int(prefix))
+                files[suffix]["files"].append({"year": int(prefix), "size": os.path.getsize(path)})
```
[the remark this answers is not in the document]
Comment (jpmckinney) [reply exporter/util.py]: Ah, just realized a list of dicts is more straightforward in plain Django, but I guess we already have `get_item` and `sortreversed`, so we can leave it for now.

**Comment by jpmckinney** [comment]
It might look better and be easier to read without the parentheses? If we need some separation, we can add an n-dash (–) between link and size. Also not sure why the space after the year is underlined.

It might be hard to read if a page has too many file sizes (kB, MB, GB, etc.). Should we stick to one (MB) or two (MB, GB)? @sabahfromlondon for advice.

**Comment by yolile** [comment]
> It might be hard to read if a page has too many file sizes (kB, MB, GB, etc.). Should we stick to one (MB) or two (MB, GB)? @sabahfromlondon for advice. Also, we can round to the nearest integer, unless it is less than 1 MB (in that case, round to one significant digit, like 0.2 MB).

Note: If we do this, we will need to implement our own size formatter (that should be a simple function anyway) and not use the default Django's one.

**Comment by sabahfromlondon** [comment]
I think I'm mostly concerned that users will not properly read that a file is in the gigabyte range before they download it. 

Because of that I would go with kB and MB for the files sizes. That way if there is a very large file size e.g. 4GB then if we state it is 4,000MB in the UI it should help to alert the user.

**Comment by yolile** [comment]
Thanks, Sabah, I will change it to kB and MB then, if you agree @jpmckinney

**Comment by jpmckinney** [comment]
I ran `find /data/storage/exporter_dumps -type f -size -1M` and we don't have anything smaller than 1MB, so let's just use MB for everything. If the result is less than 1 (maybe in the future), we can have logic to round to the nearest decimal (0.2, etc.). Otherwise, round to an integer.

**Comment by yolile** [comment]
Hmm, are you sure your command is ok? The excels from https://data.open-contracting.org/es/publication/114, for example, are less than 1MB.
image

**Comment by jpmckinney** [comment]
I guess it's not, but it was an example from the `find` docs 🤷‍♂️ 

Anyway, let's just do MB. Easier to code, and no one is worrying "Oh, 213 kB is fine, but 736 kB... I don't know".

**Comment by yolile** [comment]
It looks like this now:
image

(FYI, the extra spaces were due to the break line inside <a>)

**Comment by jpmckinney** [comment]
Looking good. "All time" still has a line break, looks like. I think removing the parentheses is easier to read. Not sure if better with an "–" n-dash (with surrounding spaces) before the size, or if it's fine with just the link (space) size.

**Comment by yolile** [comment]
image
What about this way? (without parenthesis but with a smaller font size)

**Comment by jpmckinney** [comment]
I'll check out your branch and play around with it. Hard to say what's best.

**Review by jpmckinney — APPROVED** [review]
I made some changes. Noting that if size is less than 0.1 MB, we just display 0.1 MB (instead of 0.000001 MB for a 1 byte file).

**Comment by jpmckinney** [comment]
If we want to be more correct (given imprecision for files smaller than 0.1 MB), we can display "< 1 MB" if it's less than 1 MB, instead of 0.9 MB, 0.8 MB, etc.

**Comment by jpmckinney** [comment]
Oops, forgot to update tests after last commit.

**Pull request merged**